### ISSUE TITLE
Provide an alternative to polyglot library with a library supporting all locales (e.g. intl-messageformat)

### ISSUE THREAD

**Issue by SorJEF** [marmelab/react-admin#2307]
As react-admin uses [polyglot](https://github.com/airbnb/polyglot.js) library there is no good way to use pluralization feature for many locales. For instance, this is the case with the Ukrainian and Belarussian language. Polyglot simply does not support them. Furthermore, looks like polyglot is no longer actively supported because [simple pull requests](https://github.com/airbnb/polyglot.js/pulls) adding new locales to locale groups are not getting resolved for more than a half-year.

A great solution would be to switch to another library especially as the tree shaking feature has been implemented. A candidate might be [intl-messageformat](https://github.com/yahoo/intl-messageformat), which is used across many popular and production-ready libraries like [react-intl](https://github.com/yahoo/react-intl). Furthermore, it uses official [Unicode CLDR rules for pluralization](http://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html) as well as other official Unicode specs for locale-specific formatting.

For many use cases, there is simply no alternative ways to achieve desired behavior except forking `react-admin` and reimplementing i18n. For some languages, a developer can load messages under another locale which has the same pluralization rules as the target one. But this approach cannot be used if it is required to have both the target locale and the implemented locale. This is often the case with Ukrainian, Belarussian, and other Slavic languages. In the polyglot library, there is only 1 applicable pluralization algorithm — Russian plural forms. In Ukraine, it is often required to use both languages for the app Ukrainian and Russian (because of the mostly bilingual community). Hence, described workaround is not applicable.

**Comment by SorJEF** [comment]
As I understand replacing polyglot entirely is a very difficult task. Especially considering that many translation libraries for react-admin depend on it. A good solution might be providing some way for the user to choose another formatter programmatically.

**Comment by fzaninotto** [comment]
Already discussed in https://github.com/marmelab/react-admin/issues/1099#issuecomment-332563058, and the decision we made was to not implement what you ask. See the related discussion for detailed arguments.

**Comment by kopax** [comment]
@SorJEF we made react-intl function that help you swap polyglot with react-intl

- GitHub: https://github.com/yeutech-lab/react-admin-intl
- Doc: https://yeutech-lab.github.io/react-admin-intl/

It work great, it rely on webpack alias to tweak the import and inject a different translation provider.

It exactly do :  

```js
config.resolve.alias['./i18n/TranslationProvider'] = '@yeutech-lab/react-admin-intl/lib/ra-core/i18n/TranslationProvider';
config.resolve.alias['./TranslationContext']  = '@yeutech-lab/react-admin-intl/lib/ra-core/i18n/TranslationContext';
```

I've already translated EN and FR, but more language a welcome.

This willl work as long as this package does not change the import. It will less likely happen because the translation system is already done. /cc @fzaninotto if you do I would gladly want to be informed first.
